src/requests.rs

```diff
@@ -73,3 +78,39 @@ pub fn compile(client: Client, host: &str, src: String, compiler: &str, args: St
     }
     res
 }
+
+pub fn shorten(client: Client,
+               host: &str,
+               src: String,
+               compiler: &str,
+               args: String) -> String {
+    // TODO: `compile` and `shorten` behave differently, as I was not able to
+    // pass the options that compile generates into the session options.
+    let source = json!(
+        { "sessions": [
+            {
+                "id": 1,
+                "language": "c++",
```

Comment:
I don't think this is correct, it's been a while since I've looked at this code, but shouldn't this be parameterized?

Reply:
thanks for pointing this out. This is a left-over of making it work.

Reply:
@ethanhs , fixed!